libimage/filters.go

```diff
@@ -383,6 +386,13 @@ func filterID(value string) filterFunc {
 	}
 }
 
+// filterDigest creates an digest filter for matching the specified value.
+func filterDigest(value string) filterFunc {
+	return func(img *Image) (bool, error) {
+		return string(img.Digest()) == value, nil
```

Comment:
@pmengelbert should this be `strings.HasPrefix(img.Digest(), value)?`

Reply:
No.  It's probably OK to have short IDs but a digest really should be treated as a unique identity.  `sha256:123` is incomplete.